includes/forms/class-wp-job-manager-form-submit-job.php

```diff
@@ -109,7 +109,7 @@ public function __construct() {
 			$this->step = is_numeric( $_GET['step'] ) ? max( absint( $_GET['step'] ), 0 ) : array_search( sanitize_text_field( $_GET['step'] ), array_keys( $this->steps ), true );
 		}
 
-		$this->job_id = ! empty( $_REQUEST['job_id'] ) ? absint( $_REQUEST['job_id'] ) : 0;
+		$this->job_id = ! empty( $_GET['job_id'] ) ? absint( $_GET['job_id'] ) : 0;
```

Comment:
I'll need to do more testing, but I'm worried this might break other things. I think I remember when doing the last round of sanitization I tried this and it broke something. I wonder if this is hiding a scenario where we're passing `job_id` in `POST` and in the query string but the `POST` variable is incorrect?

Reply:
The case that I identified where we are passing the `job_id` the both and the `POST` is incorrect, is when we have a preview to edit on the cookie and we try to edit another job.

What I'm thinking that we could do to keep working also with the `POST`, would be to get in the order `$_GET['job_id']` / `$_POST['job_id']` instead to trust on the EGPCS order. I don't know if we can have other problems related to this, but if you think it can be safer I can test.

What do you think?

Reply:
Hey @jom !

I added this workaround: https://github.com/Automattic/WP-Job-Manager/pull/1953/commits/f2e7c8e6b04bfa727dfa82c21c4f1f5eb13e1ea5. So if there is any case that we aren't catching or some 3rd plugin that makes something with the `POST`, we'll get. We'll just get first the `GET`. ;)